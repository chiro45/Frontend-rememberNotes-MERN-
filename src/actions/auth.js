import Swal from "sweetalert2"
import { fetchConToken, fetchSinToken } from "../helpers/fetch"
import { types } from "../types/types"


export const startLogin = (email, passwd)=>{
    return async(dispatch)=>{
        
      const resp = await fetchSinToken('auth',{email, passwd}, 'POST')
      const body = await resp.json()
      
      
      if(body.ok){
        localStorage.setItem('token',body.token )
        localStorage.setItem('token-init-date', new Date().getTime() )
        dispatch(login({
            uid: body.uid,
            name: body.name
        }))
    }else{
        Swal.fire('Error', body.msg,'error')
    }

      

    }
} 


export const startRegister = (name,email, passwd)=>{
    return async(dispatch)=>{
        const resp = await fetchSinToken('auth/register',{name, email, passwd}, 'POST')
        const body = await resp.json()
        
        if(body.ok){
            localStorage.setItem('token',body.token )
            localStorage.setItem('token-init-date', new Date().getTime() )

            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
            Swal.fire('Creado', "Usuario creado con exito",'success')
        }else{
            Swal.fire('Error', body.msg,'error')
      }
    }
}

export const startChecking = ()=>{
    return async(dispatch)=>{
        const resp = await fetchConToken('auth/renew')
        const body = await resp.json()
        if(body.ok){
            localStorage.setItem('token',body.token )
            localStorage.setItem('token-init-date', new Date().getTime() )

            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
           
        }else{
            dispatch(chekingFinish())
      }
    }
    }


const chekingFinish = ()=>({
    type: types.authChekingFinish
})


const login = (user)=>({
    type: types.authlogin,
    payload: user
})