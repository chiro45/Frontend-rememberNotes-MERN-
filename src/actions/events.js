import Swal from "sweetalert2"
import { fetchConToken } from "../helpers/fetch"
import { prepareEvents } from "../helpers/prepareEvents"
import { types } from "../types/types"

export const startAddNew = (event)=>{

    return async(dispath, getState)=>{

        const {uid, name} = getState().authReducer
        
        try {
            const resp = await fetchConToken('events',event, 'POST')
            const body = await resp.json()
           
            if(body.ok){
                event.id = body.eventoGuardado.id;
                event.user = {
                   _id: uid, 
                    name
                }
                dispath(addNew(event))
            }
            
        } catch (error) {
            console.log(error);
            Swal.fire('error','No se ha guardado','error')
        }

        

    }    


}

const addNew = (event)=>({
    type: types.eventAddnew,
    payload: event    
});


export const clearActiveNote = ()=>({
    type: types.eventClearActiveEvent
})

export const setActive = (event)=>({
    type: types.eventSetActive,
    payload: event
})

export const startEventDelete = ()=>{
    return async(dispath, getState)=>{
        const {id} = getState().calendar.activeEvent
        try {
            const resp = await fetchConToken(`events/${id}`, {}, "DELETE")
            const body = await resp.json()
             if(body.ok){
                 dispath(eventDelete())
                 Swal.fire('success','Evento Elminado','success')
             }
            
           } catch (error) {
               console.log(error)
               Swal.fire('error', 'No se pudo Eliminar el evento', 'error')
           }
        }
    }



const eventDelete = ()=>({
    type: types.eventDelete
  
})




export const eventStartLoading = ()=>{

    return async(dispath)=>{

       try {
           const resp =  await fetchConToken('events')
           const body = await resp.json()

           const events =   prepareEvents( body.eventos)

            dispath(eventLoader(events))
          
       } catch (error) {
           console.log(error)
           Swal.fire('error','No se pudieron cargar los eventos','error')
       }


    }

}

const eventLoader = (eventos)=>({
    type : types.eventLoader,
    payload : eventos
})


export const startEventUpdated = (evento)=>{
    return async(dispath)=>{
       try {
        const resp = await fetchConToken(`events/${evento.id}`, evento, "PUT")
        const body = await resp.json()
         if(body.ok){
             dispath(eventUpdated(evento))
             Swal.fire('success','Evento Actualizado','success')
         }
        
       } catch (error) {
           console.log(error)
           Swal.fire('error', 'No se pudo actualizar el evento', 'error')
       }
    }
}
const eventUpdated = (evento) =>({
    type: types.eventUpdate,
    payload: evento
})
