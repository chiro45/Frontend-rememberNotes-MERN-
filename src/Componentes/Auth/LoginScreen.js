import React from 'react'
import { useDispatch } from 'react-redux';
import Swal from 'sweetalert2';
import { startLogin, startRegister } from '../../actions/auth';
import { useForm } from '../../hooks/useForm'

export const LoginScreen = () => {
  const dispatch = useDispatch()
  const [formLoginvalues, handleLoginInputChange] = useForm({
    lEmail:'',
    lPasswd: ''
  })
  const [formRegistervalues, handleRegisterInputChange] = useForm({
    name:'',
    rEmail:'',
    rPasswd: '',
    rPasswd2: ''
  })

  const { lEmail, lPasswd} = formLoginvalues ;
  const handleLogin = (e)=>{
      e.preventDefault()
      dispatch(startLogin(lEmail, lPasswd))
  }
  const { name, rEmail, rPasswd, rPasswd2} = formRegistervalues ;
  const handleRegister = (e)=>{
    e.preventDefault()
    if(rPasswd !== rPasswd2){
      return Swal.fire('Error','Las contrase単as deben ser iguales', 'error')
    }

    dispatch(startRegister(name, rEmail, rPasswd))
    
}
  return (
    <div className='login__container'>
      <div className='container-title'>
      <h1 className='titleApp'><i className="fa-solid fa-book"></i> RememberNotes <i className="fa-solid fa-book"></i></h1>
      </div>
      
      <div className='flex-container'>
      <form onSubmit={handleLogin} className='login__login__container'>
      
      <div className='login__title__container'>
            <p>Ingresar </p>
      </div>
      <div className='login__container-inputsLogin'>
        <input className="login__inputs inputs-login"
         name='lEmail'
         value={lEmail} 
         onChange={handleLoginInputChange}
         placeholder="Email"
         type='email'/>
        <input className="login__inputs inputs-login"
          placeholder="Contrase単a"
          name='lPasswd'
         value={lPasswd} 
         onChange={handleLoginInputChange}
          type='password'/>
      </div>
      <div className='login__containerbutton'>
        <button  type='submit' className='login__button-login button'>Ingresar <i className="fa-solid fa-arrow-right"></i></button>
      </div>
      </form>
      
    <form  onSubmit={handleRegister}  className='login__register__container'>
      <div className='register__title__container'>
          <p>Registrarse</p>
      </div>
      <div className='login__container-inputsRegister'>
         <input  
         className="login__inputs inputR"
           placeholder="Nombre"
           type='text'
           name='name'
           value={name}
           onChange={handleRegisterInputChange}/>
          <input  
          className="login__inputs inputR"
            placeholder="Email"
            type='email'
            name='rEmail'
            value={rEmail}
            onChange={handleRegisterInputChange}
            />
          <input  
          className="login__inputs inputR"
            placeholder="Contrase単a"
            type='text'
            name='rPasswd'
            value={rPasswd}
            onChange={handleRegisterInputChange}
            />
          <input  
          className="login__inputs inputR"
            placeholder="Repita la contrase単a"
            type='text'
            name='rPasswd2'
            value={rPasswd2}
            onChange={handleRegisterInputChange}

            />
     </div>    
      <div className='login__containerbutton'>
          <button type='submit'className='login__button-register button'>Registrarse <i className="fa-solid fa-arrow-right-to-bracket"></i></button>
      </div>
      </form>
      </div>
</div>

   
 
 
 )
}
