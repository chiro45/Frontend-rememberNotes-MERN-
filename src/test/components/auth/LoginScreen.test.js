
import React from 'react';

import { Provider } from 'react-redux';

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import '@testing-library/jest-dom'

import { mount } from 'enzyme';

import { LoginScreen } from '../../../Componentes/Auth/LoginScreen';
import { startLogin, startRegister } from '../../../actions/auth';
import Swal from 'sweetalert2';




jest.mock('../../../actions/auth',()=>({
    startRegister: jest.fn(),
    startLogin: jest.fn()
}))

jest.mock('sweetalert2',()=>({
    fire: jest.fn()
}))


 
const middlewere = [thunk]

const mockStore = configureStore(middlewere)

const initState = {}

let store = mockStore(initState)
store.dispatch= jest.fn();

const wrapper = mount( 
    <Provider store={store} >
        <LoginScreen/>
    </Provider>
)





describe('test in <LoginScreen/>', () => {
    
    test('should be seen correctly', () => { 

        expect(wrapper).toMatchSnapshot()


     })

    test('should call the dispatch at loginScreen', () => { 

        wrapper.find('input[name="lEmail"]').simulate('change',{
            target:{
                name:'lEmail',
                value:'luciano@gmail.com'
            }
        })
        wrapper.find('input[name="lPasswd"]').simulate('change',{
            target:{
                name:'lPasswd',
                value:'1233456'
            }
        })

        wrapper.find('form').at(0).prop('onSubmit')({
            preventDefault(){}
        })
        expect( startLogin).toHaveBeenCalledWith('luciano@gmail.com','1233456')

     })


     test('should not register if passwords are diferent', () => { 


        wrapper.find('input[name="rPasswd"]').simulate('change',{
            target:{
                name:'rPasswd',
                value:'1233456'
            }
        })
        wrapper.find('input[name="rPasswd2"]').simulate('change',{
            target:{
                name:'rPasswd2',
                value:'12333456'
            }
        })

        wrapper.find('form').at(1).prop('onSubmit')({
            preventDefault(){}
        })

        expect(startRegister).not.toHaveBeenCalled()
        expect(Swal.fire).toBeCalledWith('Error','Las contrase√±as deben ser iguales', 'error')


     })
});
