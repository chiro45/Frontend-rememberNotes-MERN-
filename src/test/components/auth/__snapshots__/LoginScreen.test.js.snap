// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <LoginScreen/> should be seen correctly 1`] = `
<div
  className="login__container"
>
  <div
    className="container-title"
  >
    <h1
      className="titleApp"
    >
      <i
        className="fa-solid fa-book"
      />
       RememberNotes 
      <i
        className="fa-solid fa-book"
      />
    </h1>
  </div>
  <div
    className="flex-container"
  >
    <form
      className="login__login__container"
      onSubmit={[Function]}
    >
      <div
        className="login__title__container"
      >
        <p>
          Ingresar 
        </p>
      </div>
      <div
        className="login__container-inputsLogin"
      >
        <input
          className="login__inputs inputs-login"
          name="lEmail"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          className="login__inputs inputs-login"
          name="lPasswd"
          onChange={[Function]}
          placeholder="Contraseña"
          type="password"
          value=""
        />
      </div>
      <div
        className="login__containerbutton"
      >
        <button
          className="login__button-login button"
          type="submit"
        >
          Ingresar 
          <i
            className="fa-solid fa-arrow-right"
          />
        </button>
      </div>
    </form>
    <form
      className="login__register__container"
      onSubmit={[Function]}
    >
      <div
        className="register__title__container"
      >
        <p>
          Registrarse
        </p>
      </div>
      <div
        className="login__container-inputsRegister"
      >
        <input
          className="login__inputs inputR"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          type="text"
          value=""
        />
        <input
          className="login__inputs inputR"
          name="rEmail"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          className="login__inputs inputR"
          name="rPasswd"
          onChange={[Function]}
          placeholder="Contraseña"
          type="text"
          value=""
        />
        <input
          className="login__inputs inputR"
          name="rPasswd2"
          onChange={[Function]}
          placeholder="Repita la contraseña"
          type="text"
          value=""
        />
      </div>
      <div
        className="login__containerbutton"
      >
        <button
          className="login__button-register button"
          type="submit"
        >
          Registrarse 
          <i
            className="fa-solid fa-arrow-right-to-bracket"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
